yax/refined/src/main/scala/doobie/refined/package.scala

```diff
@@ -0,0 +1,43 @@
+package doobie
+
+package object refined {
+
+  import eu.timepit.refined.api.{Refined, Validate}
+  import eu.timepit.refined.auto._
+  import eu.timepit.refined._
+  import scala.reflect.runtime.universe.TypeTag
+
+  import doobie.imports._
+  import doobie.util.invariant._
+
+  implicit def refinedMeta[T, P](implicit metaT: Meta[T], v: Validate[T, P], tag: TypeTag[T Refined P]): Meta[T Refined P] =
```

Comment:
If you use `.nxmap` you don't have to worry about nulls and it might clean up your implementation. Let me know if you can't get it to work.